app/services/message_employee_matcher.rb

```diff
@@ -12,22 +12,24 @@ def run
   attr_reader :message
 
   def retrieve_matching_employees
-    Employee.where(started_on: day_count.days.ago)
-      .select { |employee| match_time(employee.time_zone) && message_not_already_sent?(employee) }
+    Employee.where(started_on: day_count.days.ago).select do |employee|
+      time_to_send_message?(employee.time_zone) && message_not_already_sent?(employee)
+    end
   end
 
   def day_count
     message.days_after_start
   end
 
-  def match_time(time_zone)
+  def time_to_send_message?(time_zone)
     employee_current_time = Time.current.in_time_zone(time_zone)
 
-    if employee_current_time.day == Time.current.day
-      employee_current_time_value = employee_current_time.strftime("%H%M").to_i
-      message_time_value = message.time_of_day.strftime("%H%M").to_i
-      employee_current_time_value >= message_time_value
-    end
+    return false if employee_current_time.day != Time.current.day
```

Comment:
I disagree that an `else false` is clearer than an early return. With the early return, you can tell right away which condition results in `false`. I think putting the `else false` at the end puts extra cognitive burden on the reader because they have to read through several lines of the `true` scenario, and by the time they get to the `else`, they might have forgotten what the condition at the top was.

If the `if` portion was only one line long, I wouldn't care so much about using `else`, but since the `if` portion is loaded, I think an early return is justified in this case.